components/json_encoder.rst

```diff
@@ -0,0 +1,767 @@
+            }
+
+            public function pickAWinner(): Cat
+            {
+                $jsonResource = fopen($this->catsJsonFile, 'r');
+                $type = Type::list(Type::object(Cat::class));
+
+                $cats = $this->jsonDecoder->decode($jsonResource, $type);
+
+                return $cats[rand(0, count($cats) - 1)];
+            }
+
+            /**
+             * @return list<string>
+             */
+            public function listEligibleCatNames(): array
+            {
+                $jsonString = file_get_contents($this->catsJsonFile);
+                $type = Type::list(Type::object(Cat::class));
+
+                $cats = $this->jsonDecoder->decode($jsonString, $type);
+
+                 return array_column($cats, 'name');
+            }
+        }
+
+    .. code-block:: php-standalone
+
+        // src/Service/TombolaService.php
+        namespace App\Service;
+
+        use App\Dto\Cat;
+        use Symfony\Component\JsonEncoder\DecoderInterface;
+        use Symfony\Component\JsonEncoder\JsonDecoder;
+        use Symfony\Component\TypeInfo\Type;
+
+        class TombolaService
+        {
+            private DecoderInterface $jsonDecoder;
+            private string $catsJsonFile;
+
+            public function __construct(
+                private string $catsJsonFile,
+            ) {
+                $this->jsonDecoder = JsonDecoder::create();
+            }
+
+            public function pickAWinner(): Cat
+            {
+                $jsonResource = fopen($this->catsJsonFile, 'r');
+                $type = Type::list(Type::object(Cat::class));
+
+                $cats = $this->jsonDecoder->decode($jsonResource, $type);
+
+                return $cats[rand(0, count($cats) - 1)];
+            }
+
+            /**
+             * @return list<string>
+             */
+            public function listEligibleCatNames(): array
+            {
+                $jsonString = file_get_contents($this->catsJsonFile);
+                $type = Type::list(Type::object(Cat::class));
+
+                $cats = $this->jsonDecoder->decode($jsonString, $type);
+
+                 return array_column($cats, 'name');
+            }
+        }
+
+The upper code demonstrates two different approaches to decoding JSON data
+using the JsonEncoder:
+
+* decoding from a stream (``pickAWinner``)
+* decoding from a string (``listEligibleCatNames``).
+
+Both methods work with the same JSON data but differ in memory usage and
+speed optimization.
+
+
+Decoding from a stream
+~~~~~~~~~~~~~~~~~~~~~~
+
+In the ``pickAWinner`` method, the JSON data is read as a stream using
+:phpfunction:`fopen`. Streams are useful when working with large files
+because the data is processed incrementally rather than loading the entire
+file into memory.
+
+To improve memory efficiency, the JsonEncoder creates _`ghost objects`: https://en.wikipedia.org/wiki/Lazy_loading#Ghost
```

Comment:
```suggestion
To improve memory efficiency, the JsonEncoder creates `ghost objects`_
```
And at the bottom of the file : 
.. _`ghost objects`: https://en.wikipedia.org/wiki/Lazy_loading#Ghost